workers/unity/Packages/com.improbable.gdk.core/Utility/EntityManagerExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Diagnostics;
+using System.Reflection;
+using Unity.Entities;
+#if UNITY_EDITOR
+using UnityEditor;
+
+#endif
+
+namespace Improbable.Gdk.Core
+{
+#if UNITY_EDITOR
+    // Attempt to validate the method as soon as possible when in the editor.
+    [InitializeOnLoad]
+#endif
+    internal static class EntityManagerExtensions
+    {
+        private delegate void SetComponentObjectDelegate(EntityManager entityManager,
+            Entity entity, ComponentType componentType, object componentObject);
+
+        private static readonly SetComponentObjectDelegate SetComponentObjectAction;
+
+        static EntityManagerExtensions()
+        {
+            var setComponentObjectMethodInfo =
+                typeof(EntityManager).GetMethod("SetComponentObject", BindingFlags.Instance | BindingFlags.NonPublic,
+                    null,
+                    new Type[] { typeof(Entity), typeof(ComponentType), typeof(object) },
+                    new ParameterModifier[] { });
+
+            ValidateParameters(setComponentObjectMethodInfo);
```

Comment:
This should be wrapped in an `if` `endif` I think - as the `Conditional` attribute will exclude this method from compilation

Reply:
AFAIK it turns the method call into a noop

Reply:
That's surprising - but I've never used `Conditional` before so I trust you 😄 <sup>Pls no betray me</sup>

Reply:
Can confirm what firtina says here.
This is why `Conditional` is used for loads of profiling related functions ;)